src/QsCompiler/SyntaxProcessor/ScopeTools.fs

```diff
@@ -280,3 +279,41 @@ type SymbolTracker<'P>(globals : NamespaceManager, sourceFile, parent : QsQualif
             | LocalVariable name -> decl.TypeItems.Items |> Seq.choose (namedWithName name.Value) |> Seq.toList |> function
                 | [itemDecl] -> itemDecl.Type |> StripPositionInfo.Apply
                 | _ -> addError (ErrorCode.UnknownItemName, [udt.Name.Value; name.Value]); InvalidType |> ResolvedType.New
+
+/// The context used for symbol resolution and type checking within the scope of a callable.
+type ScopeContext<'a> =
+    { /// The symbol tracker for the parent callable.
+      Symbols : SymbolTracker<'a>
+      /// True if the parent callable for the current scope is an operation.
+      IsInOperation : bool
+      /// The return type of the parent callable for the current scope.
+      ReturnType : ResolvedType
+      /// The runtime capabilities for the compilation unit.
+      Capabilities : RuntimeCapabilities
+      /// The name of the execution target for the compilation unit.
+      ExecutionTarget : QsNullable<string> }
```

Comment:
The convention right now is indeed that if something is a string  - opposed to a NonNullable<string> - then it is allowed to be null. You could hence go with just string here instead, but either works for me. A pass to get rid of the NonNullable and QsNullable types replacing them with the latest language features would indeed be good.

Reply:
I think I changed my mind and I would prefer the default value to be "Unspecified" like the Quantum SDK uses, instead of null, so I'll change this to be non-nullable.